File: Lab9/Postlab/linkedlist.c
// Ashley Nguyen - apn2my
// PostLab09 - linkedlist.c 
// 4.13.16 
// Purpose: This program reads in an int n, reads in n amount of ints and puts these values in a linked list, 
// prints out the linked list, properly deallocates the linked list 
//struct = class 


#include <stdio.h>
#include <stdlib.h>

struct ListNode{
  struct ListNode *next, *previous;
  int value; 
};

struct List {
  struct ListNode *head;
};

int main(){
  struct ListNode *head; 
  struct List *myList;
  int n;
  int num;

  //Prompt user input 
  printf( "Enter how many values to input: " );
  scanf( "%d", &n);

  myList = (struct List*) malloc(sizeof( struct List) ); //allocate list 
  myList->head = (struct ListNode*) malloc(sizeof(struct ListNode)); //allocate node 
  head = myList->head; 
  myList->head->value = 0; //initialize node's val to 0 

  for(int i = 1; i <= n; i++){
    printf( "Enter value %d: ", i );
    scanf( "%d", &num );
    head->next = (struct ListNode*) malloc(sizeof(struct ListNode));
    head->value = num;
    head = head->next; 
  }

  //print the list 
  for( int k = n; k >0; k-- ){
    printf( "%d\n", myList->head->value );
    myList->head = myList->head->next;
    } 

  //Deallocate 
  free(myList);
  return 0;
}

File: Lab12/Prelab/linkedlist.m
// Ashley Nguyen - apn2my
// PreLab12 - linkedlist.m
// 4/28/16 

#import <Foundation/NSObject.h>
#import <stdio.h>
#import <stdlib.h>
#import <math.h>

//Interface 
@interface ListNode : NSObject {
@public 
  ListNode* next; 
  int value;
}
-(int) value;
-(ListNode*) next;
@end 

//Implementation 
@implementation ListNode
-(ListNode *) next {
  return next;
}

- (int) value{
  return value;
}
@end

//main()
int main (void){
  //Dynamic allocation of listnode 
  ListNode* head = [ListNode new];
  ListNode* temp = head; 

  int input;
  printf("Enter how many values to input: ");
  scanf("%d",&input);

  //Constructing the list 
  for(int i = 0; i < input; i++){
    int val;
    printf("Enter value %d: ", i+1 );
    scanf("%d", &val);

    ListNode *linkedNode = [ListNode new];
    temp->value = val;
    temp->next = linkedNode;
    temp = temp->next; 
  }

  //Print the List 
  ListNode *print = [ListNode new];
  print = head; 

  for( int i = 0; i < input; ++i){
    int curr = print->value;
    printf("%d", curr);
    print = print->next;
    printf("\n");
  }

  //Deallocate memory 
  ListNode *dNode = [ListNode new];
  dNode = head;
 
  while( dNode != nil ){
    ListNode *removal = [ListNode new];
    removal = dNode; 
    dNode = dNode->next;
    [removal release];
  }
}
